From 374035ab26df67968de5ecfc4983f68f9b0142e4 Mon Sep 17 00:00:00 2001
From: Dan Murphy <dmurphy@ti.com>
Date: Mon, 5 Mar 2018 14:37:24 -0600
Subject: [PATCH] HACK: am335x: Work around file expansion macro

When building with the O= directive for AM335 and
an excessively long path name the build cannot fit into
the SPL region of the AM335.

The __FILE__ macro reports the full path of the file and
stores it in the data section of the image.

As a work around change the BUG() and WARN_ON macros for
the AM335 to only report the function and line number that
created the issue.  For all other devices report the filename
as normal

Signed-off-by: Dan Murphy <dmurphy@ti.com>
---
 include/linux/bug.h | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/include/linux/bug.h b/include/linux/bug.h
index f07bb716fc..cf3f735dd6 100644
--- a/include/linux/bug.h
+++ b/include/linux/bug.h
@@ -6,12 +6,26 @@
 #include <linux/compiler.h>
 #include <linux/printk.h>
 
+#ifdef CONFIG_AM33XX
+
 #define BUG() do { \
-	printk("BUG at %s:%d/%s()!\n", __FILE__, __LINE__, __func__); \
+	printk("BUG in %s():%d!\n", __func__, __LINE__); \
 	panic("BUG!"); \
 } while (0)
 
-#define BUG_ON(condition) do { if (unlikely(condition)) BUG(); } while (0)
+#define WARN_ON(condition) ({						\
+	int __ret_warn_on = !!(condition);				\
+	if (unlikely(__ret_warn_on))					\
+		printk("WARNING in /%s():%d!\n", __func__, __LINE__); \
+	unlikely(__ret_warn_on);					\
+})
+
+#else
+
+#define BUG() do { \
+	printk("BUG at %s:%d/%s()!\n", __FILE__, __LINE__, __func__); \
+	panic("BUG!"); \
+} while (0)
 
 #define WARN_ON(condition) ({						\
 	int __ret_warn_on = !!(condition);				\
@@ -19,6 +33,7 @@
 		printk("WARNING at %s:%d/%s()!\n", __FILE__, __LINE__, __func__); \
 	unlikely(__ret_warn_on);					\
 })
+#endif /* _CONFIG_AM33XX */
 
 #define WARN_ON_ONCE(condition)	({				\
 	static bool __warned;					\
@@ -31,4 +46,6 @@
 	unlikely(__ret_warn_once);				\
 })
 
+#define BUG_ON(condition) do { if (unlikely(condition)) BUG(); } while (0)
+
 #endif	/* _LINUX_BUG_H */
-- 
2.16.1.194.gb2e45c695

